src/lib/core/coordination/unique-selection-dispatcher.ts

```diff
@@ -36,9 +36,17 @@ export class UniqueSelectionDispatcher {
     }
   }
 
-  /** Listen for future changes to item selection. */
-  listen(listener: UniqueSelectionDispatcherListener) {
+  /**
+   * Listen for future changes to item selection.
+   * @return Function used to unregister listener
+   **/
+  listen(listener: UniqueSelectionDispatcherListener): Function {
     this._listeners.push(listener);
+    return () => {
+      this._listeners = this._listeners.filter((registered: UniqueSelectionDispatcherListener) => {
+        return listener !== registered;
+      });
```

Comment:
This also needs a unit test. It should set up a listener, then deregister and assert that listener isn't called again on notify